image/src/app/module/main.py

```
@@ -17,6 +44,29 @@ def module_main(data):
         [string, string]: [data, error]
     """
     try:
+        # build return body
+        if type(data) == dict:
+            return_body = processData(data)
+        else:
+            return_body = []
+            for data in data:
+                return_body.append(processData(data))
+
+        # publish data to a remote MQTT broker
+        client.publish(topic=APPLICATION['TOPIC'], payload=json.dumps(return_body), qos=APPLICATION['QOS'])
+
         return data, None
     except Exception:
         return None, "Unable to perform the module logic"
```

hiding the error is not good

---

Solved. Added a check for the MQTT broker response to verify if data transfer was successful.

---

where did you do the change?

---

Change is in lines 48 - 55